console/src/tasks.rs

```diff
@@ -11,12 +11,20 @@ use std::{
 };
 use tui::text::Span;
 
+#[derive(Debug)]
+pub(crate) enum ConnectionState {
+    Pending,
+    Connected(String),
+    Disconnected,
+}
+
 #[derive(Default, Debug)]
 pub(crate) struct State {
     tasks: HashMap<u64, Rc<RefCell<Task>>>,
     metas: HashMap<u64, Metadata>,
     last_updated_at: Option<SystemTime>,
     new_tasks: Vec<TaskRef>,
+    connection_state: ConnectionState,
```

Comment:
i'm not sure if this state belongs in the tasks module --- it's really application-wide, not specific to the task state on a runtime. although, if we ever support connecting to multiple separate remotes from one `console` process, it would make sense for this to be part of the per-remote state...but that seems a ways away...